### ISSUE TITLE
hardware_adc missing dependency on hardware_gpio

### ISSUE THREAD

**Issue by ooxi** [raspberrypi/pico-sdk#975]
When including [`hardware/adc.h`](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_adc/include/hardware/adc.h) while depending on `hardware_adc_headers` I'm getting a compile error because [`hardware/gpio.h`](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_gpio/include/hardware/gpio.h) cannot be found

```
In file included from /mysource.c++:4:
/pico-sdk/src/rp2_common/hardware_adc/include/hardware/adc.h:12:10: fatal error: hardware/gpio.h: No such file or directory
   12 | #include "hardware/gpio.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```

This is caused by neither `hardware_adc` nor `hardware_adc_headers` declaring a dependency on `hardware_gpio` / `hardware_gpio_headers`.

**Comment by kilograham** [comment]
all dependencies issues are moving to 1.5.0

**Comment by kilograham** [comment]
fixed by #1211

**Comment by kilograham** [comment]
merged into develop
